File: android/app/src/main/kotlin/com/example/gen_confi/SmartCapturePlugin.kt
package com.example.gen_confi

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.Result

class SmartCapturePlugin : FlutterPlugin, MethodChannel.MethodCallHandler {
    private lateinit var methodChannel: MethodChannel
    private lateinit var eventChannel: EventChannel
    private var nativeCameraView: NativeCameraView? = null // Holds reference to active view

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        methodChannel = MethodChannel(binding.binaryMessenger, "genconfi/mediapipe_capture")
        methodChannel.setMethodCallHandler(this)

        eventChannel = EventChannel(binding.binaryMessenger, "genconfi/mediapipe_capture_events")
        
        // Register Platform View
        binding.platformViewRegistry.registerViewFactory(
            "genconfi/mediapipe_camera_view",
            NativeCameraViewFactory(binding.binaryMessenger, eventChannel, { view ->
                nativeCameraView = view
            })
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        methodChannel.setMethodCallHandler(null)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "initialize" -> {
                // Initialization happens in the View creation mostly
                result.success(null)
            }
            "start" -> {
                nativeCameraView?.startCamera()
                result.success(null)
            }
            "stop" -> {
                nativeCameraView?.stopCamera()
                result.success(null)
            }
            "capture" -> {
                nativeCameraView?.captureImage { path, meta ->
                    if (path != null) {
                        result.success(mapOf("imagePath" to path, "meta" to meta))
                    } else {
                        result.error("CAPTURE_FAILED", "Failed to capture image", null)
                    }
                }
            }
            "dispose" -> {
                nativeCameraView?.stopCamera()
                result.success(null)
            }
            else -> result.notImplemented()
        }
    }
}

File: ios/Runner/SmartCapturePlugin.swift
// FILE: ios/Runner/SmartCapturePlugin.swift

import Flutter
import UIKit

public class SmartCapturePlugin: NSObject, FlutterPlugin {
  private var methodChannel: FlutterMethodChannel?
  private var eventChannel: FlutterEventChannel?
  private var eventSink: FlutterEventSink?
    
  // Keep track of active view(s) logic. 
  // Since PlatformViewFactory creates views, we need a way to communicate.
  // We'll use a singleton or delegate pattern for simplicity in this specific app structure.
  // Or better, the Factory passes the messenger to the view, and the view handles its own logic/events.
  // BUT the methods (start/stop) come to the Plugin. 
  // Thus we need a reference to the active view.
  
  weak var activeCameraView: NativeCameraView?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let methodChannel = FlutterMethodChannel(name: "genconfi/mediapipe_capture", binaryMessenger: registrar.messenger())
    let eventChannel = FlutterEventChannel(name: "genconfi/mediapipe_capture_events", binaryMessenger: registrar.messenger())
    
    let instance = SmartCapturePlugin()
    instance.methodChannel = methodChannel
    instance.eventChannel = eventChannel
    
    registrar.addMethodCallDelegate(instance, channel: methodChannel)
    eventChannel.setStreamHandler(instance)
    
    let factory = NativeCameraViewFactory(messenger: registrar.messenger(), plugin: instance)
    registrar.register(factory, withId: "genconfi/mediapipe_camera_view")
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initialize":
        result(nil)
    case "start":
        activeCameraView?.startCamera()
        result(nil)
    case "stop":
        activeCameraView?.stopCamera()
        result(nil)
    case "capture":
        activeCameraView?.captureImage { path, meta in
            if let path = path {
                result(["imagePath": path, "meta": meta ?? [:]])
            } else {
                result(FlutterError(code: "CAPTURE_FAILED", message: "Failed to capture", details: nil))
            }
        }
    case "dispose":
        activeCameraView?.stopCamera()
        result(nil)
    default:
        result(FlutterMethodNotImplemented)
    }
  }
}

extension SmartCapturePlugin: FlutterStreamHandler {
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        self.eventSink = events
        return nil
    }

    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        self.eventSink = nil
        return nil
    }
    
    func sendEvent(data: [String: Any]) {
        eventSink?(data)
    }
}
